# Chapter 6 - Interest Rates and Bond Valuation

Corporate Finance 9th International Edition

Ross, Westerfield, Jaffe

In [9]:
import pandas as pd

# Lets define some function so we can reuse them later.

def zero_coupon(face, ytm, years, compound=1):
    
    return face / (1 + (ytm / compound))**(years*compound)

def bond_value(face, ytm, periods, coupon, compound=1):
    
    bv = coupon * ((1 - (1/(1+(ytm/compound)))**(periods*compound)) / (ytm/compound)) + \
        face * (1 / (1 + (ytm/compound))**(periods*compound)) 
    
    return bv

**1. Valuing Bonds** - What is the price of a 10-year, zero coupon bond paying 1,000 at maturity if the YTM is:

* a. 5 percent?
* b. 10 percent?
* c. 15 percent?

In [9]:
#Given
years = 10

amount = 1000
ytms = [.05, .10, .15]

prices = []
columns = []

for yields in ytms:
    #
    price = f'{zero_coupon(amount, yields, years, compound=2):,.2f}'
    col = f'{yields*100}%'
    prices.append(price)
    columns.append(col)

df = pd.DataFrame([prices], columns=columns)
df


5.0%   10.0%   15.0%
0  610.27  376.89  235.41

**Comment**

Even though there are no coupon payments, the periods are semiannual to stay consistent with coupon bond payments.  That is why we specify compound=2.

**2. Valuing Bonds** - Microhard has issued a bond with the following characteristics:

* Par: 1,000
* Time to maturity: 25 years
* Coupon rate: 7 percent
* Semiannual payments

Calculate the price of this bond if the YTM is:
* a. 7 percent
* b. 9 percent
* c. 5 percent

In [15]:
# Given
face = 1000
periods = 25
coupon_rate = .07
compound = 2 # semiannual
coupon = (face * coupon_rate)/compound

prices = []
columns = []
ytms = [.07, .09, .05]

for y in ytms:
    
    price = f'{bond_value(face, y, periods, coupon, compound=2):,.2f}'
    col = f'{y*100:.0f}%'
    prices.append(price)
    columns.append(col)

df = pd.DataFrame([prices], columns=columns)
df

7%      9%        5%
0  1,000.00  802.38  1,283.62

**Comment**

* a. `1,000` - When the YTM and the coupon rate are equal, the bond will sell at par.
* b. `802.38` - When the YTM is greater than the coupon rate, the bond will sell at a discount.
* c. `1,283.62` - When the YTM is less than the coupon rate, the bond will sell at a premium.